internal/api/users/list_test.go

```diff
@@ -0,0 +1,135 @@
+				"course-with-lms":           {CourseID: "course-with-lms", CourseName: "Course With LMS", Role: model.GetCourseUserRole("owner")},
+				"course-without-source":     {CourseID: "course-without-source", CourseName: "Course Without Source", Role: model.GetCourseUserRole("owner")},
+				"course101":                 {CourseID: "course101", CourseName: "Course 101", Role: model.GetCourseUserRole("owner")},
+				"course101-with-zero-limit": {CourseID: "course101-with-zero-limit", CourseName: "Course 101 - With Zero Limit", Role: model.GetCourseUserRole("owner")},
+			},
+		},
+		{
+			Email: "student@test.com",
+			Name:  "student",
+			Role:  model.GetServerUserRole("user"),
+			Courses: map[string]core.EnrollmentInfo{
+				"course-languages":          {CourseID: "course-languages", CourseName: "Course Using Different Languages.", Role: model.GetCourseUserRole("student")},
+				"course-with-lms":           {CourseID: "course-with-lms", CourseName: "Course With LMS", Role: model.GetCourseUserRole("student")},
+				"course-without-source":     {CourseID: "course-without-source", CourseName: "Course Without Source", Role: model.GetCourseUserRole("student")},
+				"course101":                 {CourseID: "course101", CourseName: "Course 101", Role: model.GetCourseUserRole("student")},
+				"course101-with-zero-limit": {CourseID: "course101-with-zero-limit", CourseName: "Course 101 - With Zero Limit", Role: model.GetCourseUserRole("student")},
+			},
+		},
+		{
+			Email:   "server-admin@test.com",
+			Name:    "server-admin",
+			Role:    model.GetServerUserRole("admin"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+		{
+			Email:   "server-creator@test.com",
+			Name:    "server-creator",
+			Role:    model.GetServerUserRole("creator"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+		{
+			Email:   "server-owner@test.com",
+			Name:    "server-owner",
+			Role:    model.GetServerUserRole("owner"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+		{
+			Email:   "server-user@test.com",
+			Name:    "server-user",
+			Role:    model.GetServerUserRole("user"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+	}
+
+	response := core.SendTestAPIRequest(test, core.NewEndpoint(`users/list`), nil)
+
+	if !response.Success {
+		test.Fatalf("Response is not a success: '%v'.", response)
+	}
+
+	responseContent := response.Content.(map[string]any)
+
+	if responseContent["users"] == nil {
+		test.Fatalf("Got a nil user list.")
+	}
+
+	rawUsers := responseContent["users"].([]any)
+	actualUsers := make([]core.ServerUserInfo, 0, len(rawUsers))
+
+	for _, rawUser := range rawUsers {
+		actualUsers = append(actualUsers, *core.ServerUserInfoFromMap(rawUser.(map[string]any)))
+	}
+
+	slices.SortFunc(expectedUsers, core.CompareServerUserInfo)
+	slices.SortFunc(actualUsers, core.CompareServerUserInfo)
```

Comment:
You should not sort both your actual and expected results (in this case).
We expect the API response to be sorted, so we want to error if it is not.